src/controllers/appointment-controller.ts

```diff
@@ -0,0 +1,222 @@
+import express,{ Request, Response,NextFunction } from 'express';
+import { IAppointments } from '../models/appointment'; // Import the interface
+import AppointmentModel from '../models/appointment'; // Import the model
+import { IPatient } from '../models/patient'; // Import the interface
+import PatientModel from '../models/patient'; // Import the model
+
+//Definition of controller function for appointment
+
+export const getAllAppointmentForPatient = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const appointments = await AppointmentModel.find({ patientId: req.params.patId }).populate('patientId');
+        if (!appointments) {
+            return next({ statusCode: 404, message: 'Appointments not found' });
+        }
+        res.status(200).json(appointments);
+    } catch (error) {
+        console.log(`Error in retrieving all appointments for patient with ${req.params.patId}`, error);
+        next({ statusCode: 500, message: 'Internal server error' });
+    }
+};
+
+export const updateAppointment = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const updatedPost = await AppointmentModel.findByIdAndUpdate(
+            req.params.appointId,
+            req.body,
+            { new: true }
+        ).populate('patientId');
+        if (!updatedPost) {
+            return next({ statusCode: 404, message: 'Appointment not found' });
+        }
+        res.status(200).json(updatedPost);
+    } catch (error) {
+        next({ statusCode: 500, message: 'Internal server error' });
+    }
+};
+
+export const deleteAppointment =async (_req:Request,_res:Response,_next:NextFunction) => {
+    try {
+        const deletedPost = await AppointmentModel.findByIdAndDelete(
+            _req.params.appointId,
+            _req.body,
+        ).populate('patientId');
+        if (!deletedPost) {
+            return _next({ statusCode: 404, message: 'Appointment not found' });
+        }
+        _res.json({ message: 'Appointment deleted successfully' });
+        console.log(`Appointment with id ${_req.params.appointId} Deleted from DB`)
+      } catch (error) {
+        return _next({ statusCode: 500, message: 'Internal Server Error' });
+      }
+}
+
+export const addAppointmentToPatient = async (_req: Request, _res: Response,_next:NextFunction) => {
+    try {
+        console.log('Request Body:', _req.body); // Add this line
```

Comment:
Naming convention for params
camelCase not undercose.
Naming convenstion for function variables
camelCase if param variable is same as function variable then put underscore in the function param variable.
Naming for enum
PascalCase and starting with E
EAnimal and values will be Upper case

Reply:
Function name converstion is camelCase